dvc/repo/__init__.py

```diff
@@ -308,11 +308,11 @@ def graph(self, stages=None, from_directory=None):
                 for o in outs:
                     if o.path_info == out.path_info:
                         existing.append(o.stage)
-
-                    in_o_dir = out.path_info.isin(o.path_info)
-                    in_out_dir = o.path_info.isin(out.path_info)
-                    if in_o_dir or in_out_dir:
-                        raise OverlappingOutputPathsError(o, out)
+                    else:
```

Comment:
A bonus fix? :slightly_smiling_face:

Reply:
Could you explain what went wrong here, please? So `isin` returns true if path_infos are matching, and that caused the issue? Was it introduced by this patch, or did we have this bug earlier?

Reply:
Hm, looks like I changed `.isin()` semantics from strict to non-strict and that caused an error. This change is still desirable for optimization purposes.

Reply:
@Suor can you give a little bit more context please, what does strict vs non-strict mean?

Reply:
@shcheklein like < vs <=. It used to be `pi.isin(pi) is False` and I made it true here.

Need to revert and rename `.isin()` to make it obvious that it's strict but I can't come up with a good name.